### DISCUSSION TITLE
Can we add zoom option into zoom slider?

### DISCUSSION THREAD

**Discussion by pandika-csgi** [clientIO/joint#2170] [Q&A] [answered]
### Introduction

![image](https://user-images.githubusercontent.com/106044262/236717526-17215b48-0589-4b59-adc7-24c4ab150ca6.png)

I want to ask, if possible to add functionality like this in zoom-slider?

Thank you

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
Sure, you can add the `selectBox` type to the toolbar (https://resources.jointjs.com/docs/rappid/v3.6/ui.Toolbar.html) with all these zoom options.

**Reply by pandika-csgi** [reply]
Thanks @kumilingus, how to make them sync? (add all options and change the zoom level)
I've added `selectBox` and grouped it with the `zoom-slider`, and the result like this
![image](https://github.com/clientIO/joint/assets/106044262/750fff94-5f1f-4282-afe4-3709c7780d91)

**Reply by kumilingus** [reply, marked as answer]
Perhaps, you can use the `placeholder` of the select-box.

Tool definition:

```js
{
    type: 'selectBox',
    name: 'zoom-select',
    placeholder: 'zoom',
    options: [
        { content: '100%', value: 100 },
        { content: '60%', value: 60 },
        { content: '20%', value: 20 }
    ],
},
```

Update the placeholder on zoom:

```js
toolbar.getWidgetByName('zoom-select').selectBox.options.placeholder = '100%';
toolbar.getWidgetByName('zoom-select').selectBox.select(-1);

toolbar.on('zoom-select:option:select', function(option) {
    if (!option) return;
    paperScroller.zoom(option.value / 100, { absolute: true });
});

this.paper.on('scale', function(sx) {
    toolbar.getWidgetByName('zoom-select').selectBox.options.placeholder = Math.round(sx * 100) + '%';
    toolbar.getWidgetByName('zoom-select').selectBox.select(-1);
});
```
